app/assets/javascripts/historical.js.es6

```diff
@@ -0,0 +1,153 @@
+class HistoricalChart {
+  constructor(data, chartElement) {
+    this.data = data;
+    this.chartElement = chartElement;
+    this.dimensions = {};
+    this.filters = {};
+    this.counts = [];
+  }
+
+  filtersUpdated(data) {
+    this.data = data;
+    this.update();
+  }
+
+  render() {
+
+    function setDates() {
+      var latestDate = new Date();
+      var counterDate = new Date(2016, 1, 1);
+      var week;
+      var dates = [counterDate];
+
+      while (latestDate >= counterDate) {
+        week = counterDate.getDate() + 7;  
+        counterDate = new Date(counterDate.setDate(++week));
+        dates.push(counterDate);
+      }
+      return dates;
+    }
+
+    var datesArray = setDates();
+
+    var bookings = crossfilter(this.data);
+
+    function getCounts() {
```

Comment:
Right now it's hard for me to keep track of where these functions are in scope. If we pulled it out to the class, it might be easier to find.